### PR TITLE
Add retry logic and format code with autopep8

### PR DESCRIPTION
**Pull request by chlela**

### CODE REVIEW

**Comment by konvulse** [comment]
Frustrating.  My editor is supposed to format with Autopep8 as well.  It appears it has not been working.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+autopep8
```
Comment (konvulse) [inline requirements.txt]: Not sure autopep8 is a requirement for the project.  Seems more like something that is part of the dev env.

Comment (chlela) [inline requirements.txt]: it is recommended to keep consistent styling across the team, hence the requirement

Comment (konvulse) [inline requirements.txt]: I agree regarding the styling.  But won't a user be required to install pep8 to use the library?

Comment (chlela) [inline requirements.txt]: ohh I see what you mean. Yes yes, I agree with you

**Pull request merged**